ee/src/main/java/org/jboss/as/ee/subsystem/ManagedExecutorServiceAdd.java

```diff
@@ -105,25 +103,26 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         final AbstractManagedExecutorService.RejectPolicy rejectPolicy = AbstractManagedExecutorService.RejectPolicy.valueOf(ManagedExecutorServiceResourceDefinition.REJECT_POLICY_AD.resolveModelAttribute(context, model).asString());
 
         final ManagedExecutorServiceService service = new ManagedExecutorServiceService(name, jndiName, hungTaskThreshold, longRunningTasks, coreThreads, maxThreads, keepAliveTime, keepAliveTimeUnit, threadLifeTime, queueLength, rejectPolicy);
-        final ServiceBuilder<ManagedExecutorServiceAdapter> serviceBuilder = context.getServiceTarget().addService(ConcurrentServiceNames.getManagedExecutorServiceServiceName(name), service);
+        final ServiceBuilder serviceBuilder = context.getCapabilityServiceTarget()
+                .addService(ManagedExecutorServiceResourceDefinition.CAPABILITY.getCapabilityServiceName(name))
+                .setInstance(service);
 
-        final boolean rcPresent = context.hasOptionalCapability("org.wildfly.request-controller", null, null);
         String contextService = null;
         if(model.hasDefined(ManagedExecutorServiceResourceDefinition.CONTEXT_SERVICE)) {
             contextService = ManagedExecutorServiceResourceDefinition.CONTEXT_SERVICE_AD.resolveModelAttribute(context, model).asString();
         }
         if (contextService != null) {
-            serviceBuilder.addDependency(ConcurrentServiceNames.getContextServiceServiceName(contextService), ContextServiceImpl.class, service.getContextServiceInjector());
+            serviceBuilder.addDependency(context.getCapabilityServiceName(ContextServiceResourceDefinition.CAPABILITY.getName(), contextService, ContextServiceImpl.class), ContextServiceImpl.class, service.getContextServiceInjector());
         }
         String threadFactory = null;
         if(model.hasDefined(ManagedExecutorServiceResourceDefinition.THREAD_FACTORY)) {
             threadFactory = ManagedExecutorServiceResourceDefinition.THREAD_FACTORY_AD.resolveModelAttribute(context, model).asString();
         }
         if (threadFactory != null) {
-            serviceBuilder.addDependency(ConcurrentServiceNames.getManagedThreadFactoryServiceName(threadFactory), ManagedThreadFactoryImpl.class, service.getManagedThreadFactoryInjector());
+            serviceBuilder.addDependency(context.getCapabilityServiceName(ManagedThreadFactoryResourceDefinition.CAPABILITY.getName(), threadFactory, ManagedThreadFactoryImpl.class), ManagedThreadFactoryImpl.class, service.getManagedThreadFactoryInjector());
         }
-        if(rcPresent) {
-            serviceBuilder.addDependency(RequestController.SERVICE_NAME, RequestController.class, service.getRequestController());
+        if (context.hasOptionalCapability(REQUEST_CONTROLLER_CAPABILITY_NAME, null, null)) {
+            serviceBuilder.addDependency(context.getCapabilityServiceName(REQUEST_CONTROLLER_CAPABILITY_NAME, RequestController.class), RequestController.class, service.getRequestController());
```

Comment:
Here, you would want to use (note that serviceBuilder is a CapabilityServiceBuilder<?>):
```java
  serviceBuilder.addCapabilityRequirement(REQUEST_CONTROLLER_CAPABILITY_NAME, RequestController.class, service.getRequestController());
```

Reply:
done